src/query/parser/promql/parse.go

```diff
@@ -246,6 +273,10 @@ func (p *parseState) walk(node pql.Node) error {
 			} else {
 				if e, ok := expr.(*pql.MatrixSelector); ok {
 					argValues = append(argValues, e.Range)
+				} else if _, ok := expr.(*pql.VectorSelector); ok {
+					if variadic != 0 && n.Func.Name != linear.RoundType {
```

Comment:
Hm weird, what's special about _not_ being round type?

Reply:
It's actually more like we should only add it for date functions (which can have an arg that we don't actually care about), and it's using knowledge of prom internals (the functions with `Variadic == 1` are the date ones and the round type) which is probably a little naughty haha

Will refactor slightly to make this clear and less janky